src/libraries/System.Private.Xml.Linq/src/System/Xml/Linq/XNodeReader.cs

```diff
@@ -554,17 +554,16 @@ public override void Close()
 
         public override string GetAttribute(int index)
         {
-            // https://github.com/dotnet/runtime/issues/44287
-            //     We should replace returning null with ArgumentOutOfRangeException
-            //     In case of not interactive state likely we should throw InvalidOperationException
             if (!IsInteractive)
             {
-                return null!;
+                throw new InvalidOperationException(SR.InvalidOperation_ExpectedInteractive);
             }
+
             if (index < 0)
             {
-                return null!;
+                throw new ArgumentOutOfRangeException(nameof(index));
```

Comment:
Isn't it a breaking change? I am OK with this change, even it is documented that it will throw this exception in this case, but i guess we need to mark this PR as breaking change at least

Reply:
I've added the label so that we make sure to document this.